cpp/include/cudf/io/csv.hpp

```diff
@@ -1340,6 +1340,8 @@ class csv_writer_options {
   std::string _false_value = std::string{"false"};
   // Names of all columns; if empty, writer will generate column names
   std::vector<std::string> _names;
+  // Quote string rows, if value contains delimiter/double-quote/newline.
+  bool _quote_strings = true;
```

Comment:
Can this be an enum for the purposes of https://github.com/rapidsai/cudf/issues/7108?
It can be just two values for this PR. Unless you want to consider supporting all 4 quote-types in this PR?

Reply:
I've gone back and forth about this. I considered using `quote_style` here instead of a `bool`. What should we do if a caller uses an unexpected/unsupported `quote_style`?

(Supporting all 4 quote styles is beyond scope. :] For Hive text support, all I'm looking to do is turn off quoting without breaking anyone else.)

Reply:
I'll take another crack at using `quote_style`.

Reply:
> I've gone back and forth about this. I considered using `quote_style` here instead of a `bool`. What should we do if a caller uses an unexpected/unsupported `quote_style`?

Just don't declare more styles than you actually support?
You are coding style `QUOTE_NONE` and the current implementation is style `QUOTE_NONNUMERIC` (I think).
Only provide those two styles and default to the current one.

Reply:
Do you mean use a different enum entirely? Would that not be confusing?

Reply:
I'm not sure what you are asking. There is no enum currently. You get to make it up here.

Reply:
Oh, wait. Maybe I'm not being clear. 
We already have a `quote_style` enum in [`cudf/io/types.hpp`](https://github.com/rapidsai/cudf/blob/branch-23.02/cpp/include/cudf/io/types.hpp#L85):
```c++
enum class quote_style {
  MINIMAL,     ///< Quote only fields which contain special characters
  ALL,         ///< Quote all fields
  NONNUMERIC,  ///< Quote all non-numeric fields
  NONE         ///< Never quote fields; disable quotation parsing
};
```
I would prefer not to introduce a competing enum.

Reply:
Sorry, I did not realize that one already existed. I agree with not making up a new one.
I would then say use this one and thrown an exception on any that you are not adding support for.

Reply:
Switched now to use `quote_style`.